### PR TITLE
[INTER-726] Fix invalid host header

### PR DESCRIPTION
**Pull request by TheUnderScorer**

### CODE REVIEW

File: proxy/utils/headers.ts [hunk 1]
```diff
@@ -70,26 +70,26 @@ export function updateResponseHeaders(
...
-    logger?.verbose('Client IP resolved from x-forwarded-for', {
-      clientIp,
-      forwardedFor,
-    })
+  return stripPort(clientIp)
+}
 
-    return clientIp
+function stripPort(ip: string) {
+  if (!ip.includes(':')) {
```
Comment (bayotop) [inline proxy/utils/headers.ts:84]: Won't work as expected for IPv6 addresses.

Comment (TheUnderScorer) [reply proxy/utils/headers.ts]: Thanks, good catch! Fixed 🙂

File: proxy/utils/ip.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { isIPv4, isIPv6 } from 'net'
+
+export function stripPort(ip: string) {
```
Comment (necipallef) [inline proxy/utils/ip.ts:3]: This function is a little hard to follow. We already got `isIPv4` and `isIPv6` functions, can't we simplify the implementation using these methods, like below?

```
if (isIPv4(ip)) {
 const colonIndex = ip.indexOf(':')
 if (colonIndex === -1) {
  return ip
 }
 
 return ip.substring(0, colonIndex)
} else if (isIPv6(ip)) {
 const openingBracketIndex = ip.indexOf('[')
 const closingBracketIndex = ip.indexOf(']')
 if (openingBracketIndex === 0 && closingBracketIndex > -1) {
          return ip.substring(1, closingBracketIndex)
  }
  
  return ip
} else {
 return ip // or return empty string maybe? ip is clearly malformed
}
```

Comment (necipallef) [reply proxy/utils/ip.ts:3]: Understood 😢

**Review by necipallef — DISMISSED** [review]

**Review by necipallef — APPROVED** [review]
